sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -169,7 +169,7 @@ class Analyzer(
 
   def resolver: Resolver = conf.resolver
 
-  protected val fixedPoint = FixedPoint(maxIterations)
+  protected val fixedPoint = FixedPoint(maxIterations, errorOnExceed = true)
```

Comment:
> Analyzer should raise exception to indicates logical plan resolve failed, while optimizer just log warning to keep the current plan. 

Does exceeding max iteration indicate analyzer fails logical plan resolving? Doesn't it just mean the plan is not fixed after max iterations? It sounds not equal.

Reply:
Yeah, I see. I've updated the descriptions of PR & JIRA. Thanks for pointed it out :-)